op-batcher/batcher/channel_manager_test.go

```diff
@@ -318,6 +318,47 @@ func newFakeDynamicEthChannelConfig(lgr log.Logger,
 	}
 }
 
+// TestChannelManager_PublishingBacklog tests that the channel manager will not time out
+// when publishingBacklog is set to true in the signal struct.
+func TestChannelManager_PublishingBacklog(t *testing.T) {
+	l := testlog.Logger(t, log.LevelCrit)
+
+	cfg := channelManagerTestConfig(10000, derive.SingularBatchType)
+	cfg.MaxChannelDuration = 20
+	cfg.InitNoneCompressor()
+
+	m := NewChannelManager(l, metrics.NoopMetrics, cfg, defaultTestRollupConfig)
+
+	// Seed channel manager with blocks
+	rng := rand.New(rand.NewSource(99))
+	for range 2 {
+		block := derivetest.RandomL2BlockWithChainId(rng, 2, defaultTestRollupConfig.L2ChainID)
+		m.blocks.Enqueue(SizedBlock{Block: block})
+	}
+
+	// Call TxData a first time - if `publishingBacklog` is `false`, channel would be timed out.
+	_, err := m.TxData(eth.BlockID{Number: 21}, false, false, pubInfo{forcePublish: false, publishingBacklog: true})
+	require.ErrorIs(t, err, io.EOF)
```

Comment:
If the channel is timed out, why does it return `io.EOF`? If the comment is correct, wouldn't we want to do the same assertions as in L354-359?